.github/workflows/build-and-test.yaml

```diff
@@ -2,8 +2,11 @@ name: Build-and-test
 
 on:
   push:
+    branches:
```

Comment:
Ensure that the 'types' key under 'pull_request' is intentionally removed. If specific pull request types should trigger the workflow, such as 'opened' or 'synchronize', they should be explicitly listed. [important]